astropy/modeling/tests/test_fitters.py

```diff
@@ -21,21 +21,16 @@
 from astropy.utils.data import get_pkg_data_filename
 from astropy.stats import sigma_clip
 
+from astropy.utils.compat.optional_deps import HAS_PKG_RESOURCES, HAS_SCIPY
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.modeling.fitting import populate_entry_points
 from . import irafutil
 
-try:
+if HAS_SCIPY:
     from scipy import optimize
-    HAS_SCIPY = True
-except ImportError:
-    HAS_SCIPY = False
 
-try:
+if HAS_PKG_RESOURCES:
```

Comment:
This one might conflict with astropy/astropy#11091 . How to resolve the conflict depends on which PR gets merged first. ⚔️  cc @saimn